src/components/PostsList.tsx

```diff
@@ -7,13 +7,13 @@ import { Post } from '../types/Post';
 type Props = {
   posts: Post[];
   selectedPostId?: number;
-  onPostSelected: (post: Post | null) => void;
+  setSelectedPost: (newPost: Post | null) => void;
```

Comment:
The type definition for setSelectedPost should accept a Post or null, but it should actually accept a Post's id (number) or null, as the function is called with post.id in line 43.